packages/app/src/stores/AgoraStore/AgoraScreenShareStore/AgoraScreenShareStore.ts

```diff
@@ -57,26 +57,31 @@ const AgoraScreenShareStore = types
     }
   }))
   .actions((self) => ({
-    handleUserPublished: flow(function* (user: IAgoraRTCRemoteUser, mediaType: 'audio' | 'video') {
+    handleUserPublished: flow(function* (
+      user: IAgoraRTCRemoteUser,
+      mediaType: 'audio' | 'video',
+      callback: () => void
+    ) {
       yield self.client.subscribe(user, mediaType);
 
       self.remoteVideoTrack = user.videoTrack;
+      callback();
     }),
     handleUserUnpublished() {
       self.remoteVideoTrack?.stop();
       self.remoteVideoTrack = undefined;
     }
   }))
   .actions((self) => ({
-    init(worldId: string, userId: string) {
+    init(worldId: string, userId: string, callback: () => void) {
       self.appId = appVariables.AGORA_APP_ID;
       self.worldId = worldId;
-      this.setupAgoraListeners();
+      this.setupAgoraListeners(callback);
       self.openScreenShare(userId);
     },
-    setupAgoraListeners() {
+    setupAgoraListeners(callback: () => void) {
```

Comment:
```suggestion
    setupAgoraListeners(onUserPublished: () => void) {
```
cause what if we add in future also a callback on user-unpublished? also it would be clear what is the callback for